baselines/pFedHN/README.md

```diff
@@ -0,0 +1,96 @@
+
+## About this baseline
+
+****What’s implemented:**** "The code in the repository reproduces the paper by implementing the concept of hypernetworks which create weights for that target network thus resolving the problems of dataheterogenity.The hypernetworks lies in the server and the clients have the target model. Initially hypernetwork sends the weights from the server which is loaded into the target net . After the targetnet is trained we pass the delta_theta to the client , which inturn updated the phi_gradients for the hypernetwork and the flow goes on."
+
+****Datasets:**** MNIST,CIFAR10,CIFAR100 from torchvision 
+
+****Hardware Setup:**** Will be updated
+
+****Contributors:**** :Ram Samarth B B(@achiverram28) , Kishan Gurumurthy(@kishan-droid) , Sachin DN(@sachugowda)
+
+
+## Experimental Setup
+
+****Task:**** : Image Classification
+
+****Model:**** : CNNHyper for the HyperNetwork , CNNTarget For the TargetNetwork
+
+**Dataset:** This baseline includes the MNIST, CIFAR10 , CIFAR100 datasets. By default it will be partitioned into 50 clients following IID distribution. The settings are as follow:
+
+| Dataset | #classes | partitioning method | classes per client |
+| :------ | :---: | :---: | :---: |
+| MNIST | 10 | Non-IID | 2 |
+| CIFAR10 | 10 | Non-IID | 2 |
+| CIFAR10 | 100 | Non-IID | 10 |
+
+
+****Training Hyperparameters:**** The following table shows the main hyperparameters for this baseline with their default value (i.e. the value used if you run `python3 main.py` directly)
+
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 50 |
+| local epoch | 50 |
+| batch size | 64 |
+| number of rounds | 5000 |
+| HyperNetwork learning rate | 1e-2 |
+| TargetNetwork learning rate | 5e-3 |
+| TargetNetwork weight decay | 5e-5 |
+| TargetNetwork momentum | 0.9 |
+| data partition | Non - IID |
+| HyperNetwork Optimizer | Adam |
+| TargetNetwork Optimizer | SGD with weight decay |
+**Model variations** : The model is flexible to use for images with 1 channel
+and 28x28 dimension , images with 3 channels and 32X32 dimension(for MNIST,CIFAR10 and CIFAR)
```

Comment:
You mention `CIFAR-10` and `CIFAR` ... did you want to say `CIFAR-100`?

Reply:
Also probably is best to move the `MNIST` next to the "28x28 dimension" part of the sentence to avoid confusion since that's the input size for MNIST.